### PR TITLE
Notification fixes and enhancements.

### PR DESCRIPTION
**Pull request by kadams54**

### CODE REVIEW

**Comment by kadams54** [comment]
## QA

1. Shrink the screen size and verify that overlap and z-index seems reasonable/usable.
2. Close the error notification. Make sure that the closing animation is not janky.
3. Do stuff that triggers notifications (e.g., deploy a bundle) and verify that they look OK in the top position.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4225/
Test PASSed.

File: jujugui/static/gui/src/app/components/notification-list/_notification-list.scss [hunk 1]
```diff
@@ -1,7 +1,10 @@
 .notification-list {
+    $width: 45%;
     position: absolute;
-    bottom: $panel-space;
-    left: $panel-space;
-    max-width: 50%;
+    top: 0;
+    left: 50%;
+    margin: 0;
+    margin-left: -1 * ($width / 2);
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/notification-list/_notification-list.scss]: :heart: this! The correct way to case in Sass.

**Comment by anthonydillon** [comment]
QA (with lucas comment about alignment) and code :+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4231/
Test PASSed.

**Comment by hatched** [comment]
This looks great, good improvement. My only little issue is that the animate 'up' is a little janky when there are multiple messages:

Every second or so run `app.db.notifications.add({ message: 'hey' })` in the console and you can see the 'stepped' approach it takes when animating up

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4233/
Test PASSed.

**Comment by hatched** [comment]
You increased the duration of the 'appear' as well as the 'fade' which makes the notification not show up for a few seconds :)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4244/
Test PASSed.

**Comment by huwshimi** [comment]
The slide up bit looks good in Chrome, Firefox leaves an extra gap after the notification slides up, but that can be a followup as discussed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4251/
Test FAILed.

**Comment by hatched** [comment]
:+1: QA :ok: 
Thanks for the updates!

**Comment by kadams54** [comment]
Thanks all! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4252/
Test PASSed.

**Pull request merged**